docs/labels.md

```diff
@@ -11,50 +10,93 @@ key/value labels set on it, with at most one label with a particular key.
 }
 ```
 
-While there are no restrictions on the format of label values, label keys must be of the form:
-```
-label-key ::= prefixed-name | name
-prefixed-name ::= prefix '/' name
-prefix ::= DNS_SUBDOMAIN
-name ::= DNS_LABEL
-```
-DNS_LABEL and DNS_SUBDOMAIN are defined in the [identifiers design doc](/docs/design/identifiers.md). The prefix is optional. If the prefix is not specified, the key is assumed to be private to the user. Other system components that wish to use labels must specify a prefix. The "kubernetes.io/" prefix is reserved for use by kubernetes components.
+## Motivation
 
-Unlike [names and UIDs](identifiers.md), labels do not provide uniqueness. In general, we expect many objects to carry the same label(s). 
+Service deployments and batch processing pipelines are often multi-dimensional entities (e.g., multiple partitions or deployments, multiple release tracks, multiple tiers, multiple micro-services per tier). Management often requires cross-cutting operations, which breaks encapsulation of strictly hierarchical representations, especially rigid hierarchies determined by the infrastructure rather than by users. Labels enable users to map their own organizational structures onto system objects in a loosely coupled fashion, without requiring clients to store these mappings.
```

Comment:
Please move the last sentence (Labels enable users...) first, and put it in a paragraph on its own.